openquake/risklib/riskinput.py

```diff
@@ -659,56 +652,29 @@ class GmfDataGetter(GmfGetter):
     """
     Extracts a dictionary of GMVs from the datastore
     """
-    def __init__(self, gmf_data, grp_id, rlzs_by_gsim, start=0, stop=None):
+    def __init__(self, gmf_data, start=0, stop=None):
         self.gmf_data = gmf_data
-        self.grp_id = grp_id
-        self.rlzs_by_gsim = rlzs_by_gsim
         self.start = start
         self.stop = stop
-        self.gmf_data_dt = gmf_data[next(iter(gmf_data))].dtype
 
     def init(self):
         pass
 
     def gen_gmv(self):
         """
-        Yield gmv records from the datastore, if present
+        Returns gmv records from the datastore, if any
         """
-        key = 'grp-%02d' % self.grp_id
-        try:
-            dset = self.gmf_data[key]
-        except KeyError:
-            return
-        for rec in dset[self.start:self.stop]:
-            yield rec
+        return self.gmf_data['data'][self.start:self.stop]
```

Comment:
The name `gen_gmv` has to stay, since it is the name of the API (all getters have a `gen_gmv` method). Sometimes it is a generator, sometimes it is not.